### DISCUSSION TITLE
Config & Decorator

### DISCUSSION THREAD

**Discussion by koayst** [explosion/spaCy#9352] [Help: Other Questions] [answered]
Hi,

Trying to learn the new way of training a model under Spacy using the CFG file.  While looking at the CFG file from some sample codes published in github, notice this "@xxxx" decorator.  Understand it is pointing to a "function".  

Trying to find the documentation, for example, "@architectures", "@layers", "@misc", "@schedules", "@readers", "@batchers" ... etc.  

Can you please point me to the link/place for me to read up ?  Thanks.

**Comment by rspeer** [comment, marked as answer]
The configuration items starting with `@` indicate that, instead of a static value, the value is being provided by a function. This idea is introduced in the documentation at https://spacy.io/usage/training#config-functions .

The example there uses a function to provide a `batch_size` that changes according to a schedule, instead of being a fixed value.

The [function registry](https://spacy.io/api/top-level#registry) section of the documentation lists what kinds of functions can be registered.

(I'm new to the spaCy team and this is the first question I'm answering, so I hope my explanation is right!)

**Reply by koayst** [reply]
@rspeer Thank you so much.  The write-ups under [function registry](https://spacy.io/api/top-level#registry) was what I was searching high low.
